RIOT-OS/RIOT
# cpu/kinetis: implement Low-Leakage Wakeup Unit and add support to UART

### Contribution description
Split out of #11789 and originally partly from #7897, this PR adds support for the Kinetis LLWU peripheral (which is used by other peripherals to wake from the lowest power mode) and adds UART support to make use of it.

This allows sleeping all the way down to the lowest power mode (~5uA) while retaining a working LEUART or even UART by waking on a pin interrupt instead of waking via the UART hardware.


### Testing procedure
Compile with `CFLAGS=-DPM_BLOCKER_INITIAL=0x0 make` to unblock all modes by default and confirm that the shell remains working.

It should remain working either (depending on the baud rate and board clock configuration) because the correct power mode has been blocked to keep the UART clocked or because it is able to wake via LLWU or GPIO interrupt, but in any case it should "automatically" remain working for any baud and board configuration.

_Currently there are bugs preventing GPIO interrupts from working properly on Kinetis. This is fixed in #14376._

_There also may be an issue with some boards working at baud rates below 115200 which appears to be reproducible on master._


### Issues/PRs references
This PR needs #14376 for GPIO interrupt bugfixes.
